src/xenia/app/emulator_window.cc

```diff
@@ -928,6 +936,41 @@ void EmulatorWindow::ToggleFullscreen() {
   SetFullscreen(!window_->IsFullscreen());
 }
 
+void EmulatorWindow::TakeScreenshot() {
+  xe::ui::RawImage myimage;
+  if (emulator()->is_title_open()) {  // Make sure we have a game open first
+    GetGraphicsSystemPresenter()->CaptureGuestOutput(myimage);
+    ExportImage(myimage);
+  }
+}
+
+void EmulatorWindow::ExportImage(const xe::ui::RawImage& image) {
+  auto path = std::filesystem::path("screenshot.png");
```

Comment:
The path can be passed as a temporary, without an intermediate variable. Though overall it's better to add some date and time (and also some resolution for multiple screenshots taken during the same second) instead of overwriting the previous scerenshot.